File: frontend/src/interfaces/Usuario.ts
export interface Usuario {
    id: number;
    nombres: string;
    apellidos: string;
    rut: number;
    dv: string;
    fechaNacimiento: string;
    correoElectronico: string;
    contrasena: string;
}

export type UsuarioCreate = Omit<Usuario, 'id'>;

File: api/src/main/java/com/previred/api/entities/Usuario.java
package com.previred.api.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Entity
@Table(name = "usuarios")
@Data
@NoArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombres;

    @Column(nullable = false)
    private String apellidos;

    @Column(nullable = false, unique = true)
    private Long rut;

    @Column(nullable = false)
    private String dv;

    @Column(nullable = false)
    private Date fechaNacimiento;

    @Column(nullable = false, unique = true)
    private String correoElectronico;

    @Column(nullable = false)
    private String contrasena;
}
